tests/integration/kyberPriceFeed.test.js

```diff
@@ -0,0 +1,313 @@
+/*
+ * @file Tests simple cases of updating price against a Kyber deployment
+ *
+ * @test Delegated updater can update the feed
+ * @test MLN price update on reserve changes price on feed post-update
+ * @test Normal spread condition from Kyber rates yields midpoint price
+ * @test Crossed market condition from Kyber rates yields midpoint price
+ * @test In backupUpdate, assets must match registry and WETH price must be 1 ether
+ * @test Only registry owner can use emergency update function
+ */
+
+import { BN, toWei } from 'web3-utils';
+import { BNExpDiv, BNExpInverse } from '~/tests/utils/BNmath';
+import { CONTRACT_NAMES } from '~/tests/utils/constants';
+import { partialRedeploy } from '~/deploy/scripts/deploy-system';
+import { deploy, send, call } from '~/deploy/utils/deploy-contract';
+import web3 from '~/deploy/utils/get-web3';
+
+let deployerTxOpts, updaterTxOpts;
+let conversionRates, kyberPriceFeed, kyberNetworkProxy, registry;
+let deployer, updater;
+let eur, mln, weth, registeredAssets;
+
+beforeAll(async () => {
+  const accounts = await web3.eth.getAccounts();
+  [deployer, updater] = accounts;
+  deployerTxOpts = { from: deployer, gas: 8000000 };
+  updaterTxOpts = { from: updater, gas: 8000000 };
+  const deployed = await partialRedeploy([CONTRACT_NAMES.VERSION]);
+  const contracts = deployed.contracts;
+
+  eur = contracts.EUR;
+  mln = contracts.MLN;
+  weth = contracts.WETH;
+  conversionRates = contracts.ConversionRates;
+  kyberNetworkProxy = contracts.KyberNetworkProxy;
+
+  registry = await deploy(CONTRACT_NAMES.REGISTRY, [deployer], deployerTxOpts);
+
+  kyberPriceFeed = await deploy(
+    CONTRACT_NAMES.KYBER_PRICEFEED,
+    [
+      registry.options.address,
+      kyberNetworkProxy.options.address,
+      toWei('0.5', 'ether'),
+      weth.options.address,
+      toWei('0.1', 'ether')
+    ],
+    deployerTxOpts
+  );
+  await send(registry, 'setNativeAsset', [weth.options.address], deployerTxOpts);
+
+  for (const addr of [eur.options.address, mln.options.address, weth.options.address]) {
```

Comment:
Aren't these already registered during deployment?

Reply:
Ah, I see that we're deploying a new registry above. Is that essential, or just legacy code from the mock tests?

Reply:
This was mainly to simplify the arguments when we call `update`. When we deploy Registry regularly there are like 10 assets or something, but we only need a few to test the functionality.